lib/chef/knife/kvm_vm_create.rb

```diff
@@ -140,6 +144,15 @@ def read_and_validate_params
           exit 1
         end
 
+        if (config[:location].nil? &&
```

Comment:
I would use XOR because that's exactly what it's intended use is.  I don't think it's a comparator that engineers wouldn't understand.